src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -450,79 +529,150 @@ def pivot(
             ------------------------------
             <BLANKLINE>
         """
-        self._df, pc, pivot_values, default_on_null = prepare_pivot_arguments(
+
+        self._df, pc, pivot_values, pivot_default_on_null = prepare_pivot_arguments(
```

Comment:
I know we made an earlier effort to replace `self._df` -> `self._dataframe`? Shall we do it here as well?